src/v/resource_mgmt/memory_groups.cc

```diff
@@ -94,16 +103,21 @@ size_t system_memory_groups::total_memory() const {
 
 system_memory_groups& memory_groups() {
     static thread_local std::optional<system_memory_groups> groups;
-    if (!groups) {
-        size_t total = ss::memory::stats().total_memory();
-        bool wasm = wasm_enabled();
-        if (wasm) {
-            size_t wasm_memory_reservation
-              = config::shard_local_cfg()
-                  .wasm_per_core_memory_reservation.value();
-            total -= wasm_memory_reservation;
-        }
-        groups.emplace(total, wasm);
+    if (groups) {
+        return *groups;
+    }
+    size_t total = ss::memory::stats().total_memory();
+    bool wasm = wasm_enabled();
+    const auto& cfg = config::shard_local_cfg();
+    if (wasm) {
+        size_t wasm_memory_reservation
+          = cfg.wasm_per_core_memory_reservation.value();
+        total -= wasm_memory_reservation;
+    }
+    compaction_memory_reservation compaction;
+    if (cfg.log_compaction_use_sliding_window.value()) {
+        compaction.max_bytes = cfg.storage_compaction_key_map_memory.value();
```

Comment:
why do we adjust total based on wasm size here, but adjust for compaction in the system memory group constructor?

Reply:
I'd originally done the accounting here, but opted to move it into `system_memory_groups` after I fudged the values a couple times. I can move it back, or move wasm accounting over to `system_memory_groups` too, if we wanna be consistent

Reply:
i don't have a strong opinion one way or another about it

Reply:
Left this as is